### PR TITLE
Update Slovenia holidays: add 2023 Solidarity Day

### PR DESCRIPTION
**Pull request by kanobi**
## Proposed change

This PR adds the "Solidarity Day" public holiday for Slovenia for August 14th 2023. 
https://en.wikipedia.org/wiki/Public_holidays_in_Slovenia

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Version 0.31
+============
+
+Released ???????? ??, ????
+
```
Comment (PPsyrius) [inline CHANGES:5]: You might want to file this PR against `beta` branch instead of `master`

File: holidays/countries/slovenia.py [hunk 2]
```diff
@@ -80,6 +80,10 @@ def _populate(self, year):
         # Independence and Unity Day.
         self._add_holiday_dec_26(tr("dan samostojnosti in enotnosti"))
 
+        if year == 2023:
+            # Solidarity Day. Only in 2023.
+            self._add_holiday_aug_14(tr("dan solidarnosti"))
+
```
Comment (PPsyrius) [inline holidays/countries/slovenia.py:86]: For one-off holidays like this, using `special_holiday` might be better (see [Argentina](https://github.com/vacanza/python-holidays/blob/master/holidays/countries/argentina.py) and [Singapore](https://github.com/vacanza/python-holidays/blob/master/holidays/countries/singapore.py)'s implementation, for example) as well as some quick testcases.
```suggestion
    special_holidays = {
        # Solidarity Day. Only in 2023.
        2023: (AUG, 14, tr("dan solidarnosti")),
    }
```

**Review by PPsyrius — COMMENTED** [review]
Some suggestions here :)

File: holidays/countries/slovenia.py (hunk 2 above)
Comment (kanobi) [reply holidays/countries/slovenia.py]: This is more clearer yes. Thanks!

**Review by KJhellico — COMMENTED** [review]
LGTM.

**Comment by arkid15r** [comment]
Thanks for the contribution and prompt comments addressing @kanobi!

I've just merged it into beta (don't worry about the failed tests, it's unrelated to your PR)

**Pull request merged**